### PR TITLE
Hide eol distros

### PR DESCRIPTION
**Pull request by mikaelarguedas**
This PR is a replacement for previously reverted #169

- uses loaded distributions to generate Package and Stack headers
- uses the already existing list of active distributions (distro_names_buildfarm) to create the ros version selector

Pros:
- remove modification of imported "distro_names" in all modules using it
- display only active distros on all wiki pages
- keep compatibility with macros passed argument using eol distros (e.g. groovy_and_newer), fixe #197 

Cons:
- no way for user to access wiki pages for not displayed distros

Alternative:
Create a checkbox "display EOL distros" next to the version selector that adds back all distros to the version selector and allow to access content for all of them

### CODE REVIEW

**Comment by tfoote** [comment]
In the meeting we talked about refactoring this to hide the EOL'd distro selectors, but not completely disable them.

**Comment by mikaelarguedas** [comment]
Delaying this until less radical approach is implemented

**Comment by dhood** [comment]
hey @mikaelarguedas and/or @wjwwood @tfoote, how were you testing this PR?

I have a local instance of the wiki, and was going to create a test page that uses that macro, but first I have to create a user and I haven't been able to do that successfully yet (haven't tried disabling the whitelist yet)

Did you test on a local instance, or just on the live one?

**Comment by mikaelarguedas** [comment]
I was testing live for short time periods

**Comment by dhood** [comment]
ok, thanks!

**Comment by dhood** [comment]
- Now there's a checkbox that is unchecked by default but if checked, the version selector for eol distros appears.

- content-embedding macros for all distros are supported (e.g. diamondback_and_newer can be used even though it doesn't have a version selector).

- If the distro selected in the url is an EOL distro, the eol selector is displayed on page load.

- If a package has only been released into eol distros, the checkbox isn't there.


example stack header with only some distros released (eol distros are hidden by default)
package_header

---

example stack header with only EOL distros released
package_header_eol_only

---

example Version() (eol distros are hidden by default)
version_macro

File: macro/macroutils.py [hunk 1]
```diff
@@ -14,6 +14,8 @@
 distro_names = ['boxturtle', 'cturtle', 'diamondback', 'electric', 'fuerte', 'groovy', 'hydro', 'indigo', 'jade', 'kinetic', 'lunar', 'melodic', 'unstable']
 distro_names_indexed = ['diamondback', 'electric', 'fuerte', 'groovy', 'hydro', 'indigo', 'jade', 'kinetic', 'lunar', 'melodic', 'unstable'] #boxturtle and cturtle not indexed
 distro_names_buildfarm = ['indigo', 'jade', 'kinetic', 'lunar', 'melodic']
+distro_names_never_on_buildfarm = ['boxturtle', 'cturtle', 'diamondback', 'unstable']
```
Comment (dhood) [inline macro/macroutils.py]: I just took a guess at why these distros are being excluded from the macros, open to suggestions for a more appropriate variable name if there is one

**Comment by dhood** [comment]
With jade removed from the non-EOL list, Version header looks like:

by default:
version_jade_eol_collapsed

expanded:
version_jade_eol

**Comment by dhood** [comment]
I actually think that cturtle and diamondback have been removed from the version selector by accident somewhere along the line (@mikaelarguedas pointed out all files are modifying the global `distro_names`, so that would do it).

Our ubuntu installation page has code to load the cturtle/diamondback installation pages: http://wiki.ros.org/Installation/Ubuntu (which it can't do currently because the version selector starts at electric).

I definitely think that we should restore support for macros referencing older distros e.g. http://wiki.ros.org/image_transport has a section using `diamondback_and_newer` that is not currently being displayed. This PR does that. But, do we also want to add cturtle and diamondback back to the (EOL) version selector?

---

btw this is the contents of the wiki page I previously posted screenshots for, which is what's making the text show up:

```
<<Version()>>

{{{{#!wiki version kinetic
$ROS_DISTRO
}}}}

{{{{#!wiki version diamondback_and_newer
as of diamondback
}}}}
{{{{#!wiki version indigo_and_newer
as of indigo
}}}}
```
And the following for PackageHeader (I'm testing on a dummy wiki which is why it gives index errors):
```
<<PackageHeader(rviz)>>

{{{{#!wiki version kinetic
$ROS_DISTRO
}}}}

{{{{#!wiki version diamondback_and_newer
`my_api()` <<Version(ROS diamondback)>>
}}}}
```

**Comment by dhood** [comment]
This PR now highlights that EOL distros are EOL.


Version() macro:

version_macro_highlight_eol

PackageHeader:

package_header_eol_only2

**Comment by dhood** [comment]
@tfoote @mikaelarguedas @wjwwood could any of you (re)review this if you get a chance

**Pull request merged**